recipes/azure-sdk-for-cpp/all/conanfile.py

```diff
@@ -105,9 +138,29 @@ def package_info(self):
         # core component
         self.cpp_info.components["azure-core"].set_property("cmake_target_name", "Azure::azure-core")
         self.cpp_info.components["azure-core"].libs = ["azure-core"]
-        self.cpp_info.components["azure-core"].requires.extend(["openssl::openssl", "libcurl::curl", "libxml2::libxml2"])
 
-        enabled_sdks = [sdk for sdk in AZURE_SDK_MODULES if self.options.get_safe(sdk)]
-        for sdk in enabled_sdks:
-            self.cpp_info.components[sdk].set_property("cmake_target_name", f"Azure::{sdk}")
-            self.cpp_info.components[sdk].libs = [sdk]
+        self.cpp_info.components["azure-core"].requires.extend(["openssl::openssl", "libxml2::libxml2"])
+
+        if self.settings.os == "Windows" and self.options.get_safe("win_http_transport"):
+            self.cpp_info.components["azure-core"].requires.append("wil::wil")
+        else:
+            self.cpp_info.components["azure-core"].requires.append("libcurl::libcurl")
```

Comment:
```suggestion
            self.cpp_info.components["azure-core"].system_libs = ["winhttp"]
        if self.options.curl_http_transport:
            self.cpp_info.components["azure-core"].requires.append("libcurl::libcurl")
```

On windows, needs [winhttp](https://learn.microsoft.com/en-us/windows/win32/winhttp/iwinhttprequest-interface) system library too. Otherwise, it will fail with:

> azure-core.lib(win_http_transport.cpp.obj) : error LNK2019: unresolved external symbol __imp_WinHttpSendRequest referenced

Keep curl as optional always.

Reply:
>  I'm guessing that's if you're making a custom cmake for your app and assuming it will be brought in automatically as a public library using the conan toolchain?

Not really, only running regular Conan create command and breaking in the test package.